docs/src/implemented-proposals/multi-level-multi-iterator-scheduler.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Multi-Level Multi-Iterator Scheduler
+---
+
+## Background
+
+We have proposed graph-based scheduling in the past, however, as a hold-over until we implement those the current scheduler has been modified to use a multi-iterator approach. This approach significantly improves single-threaded throughput in situations with high-contention - and lends itself to a straight-forward stepping stone to graph-based scheduling, with a few modifications.
+
+## Single-Level Multi-Iterator
+
+The single-level multi-iterator has already been implemented in Solana's independent banking stage thread schedulers, this section serves to outline that concept in order to provide context for the multi-level multi-iterator.
+
+First, we take our priority-queue and collect the entire buffer into a vector. We initialize multiple, in this case 64, positions such that there are no write-conflicts between the transactions at the given positions. These, up to 64, transactions then make up a transaction batch to be executed. The multi-iterator marks these transactions as `already_processed`, and the positions are marched forward, again checking for conflicts as we set them.
+
+### Example
+
+For simplicity's sake, let's assume we have a batch-size of 2 instead of 64, and a priority-queue with the following transactions:
+
+```text
+[A, A, A, B, C, A, B, C, D]
+ ^        ^                     // initial iterator positions
+```
+
+The first batch would be `[A, B]`. Marching the positions forward, we get:
+
+```text
+[A, A, A, B, C, A, B, C, D]
+    ^        ^
+```
+
+and the second batch is `[C, D]`. Marching the positions forward again, we get:
+
+```text
+[A, A, A, B, C, A, B, C, D]
+       ^           ^
+```
+
+giving a batch of `[A, B]`. Note that the second iterator skipped over an `A` transaction, as that transaction would conflict with the `A` our first position is pointing to. Following this the remaining batches would be `[[A, C], [D]]`.
```

Comment:
Would be helpful to add a quick section on when the skipped transactions are retried, I'm assuming the iterator resets? Diagram for that reset case, where we ignore the already processed transactions would be helpful too

Reply:
Added the overall batches for this iteration process to make it a bit more clear that all the txs get selected. Added a note that for retryable errors, they are re-inserted at the end of this process (same as what happens now).